crates/rpc/rpc-types-compat/src/proof.rs

```diff
@@ -6,7 +6,11 @@ use reth_trie_common::{AccountProof, StorageProof};
 
 /// Creates a new rpc storage proof from a primitive storage proof type.
 pub fn from_primitive_storage_proof(proof: StorageProof) -> EIP1186StorageProof {
-    EIP1186StorageProof { key: JsonStorageKey(proof.key), value: proof.value, proof: proof.proof }
+    EIP1186StorageProof {
+        key: JsonStorageKey::Hash(proof.key),
+        value: proof.value,
+        proof: proof.proof,
+    }
 }
```

Comment:
Idea for followup, should we move those alloy types into alloy-eips?

Reply:
eip1186 mod might make sense, though this EIP is basically just RPC spec so probably would still be closer to RPC types